myia/basics.py

```diff
@@ -0,0 +1,131 @@
+"""Global implementations useful for compiled modules."""
+from ovld import ovld
+
+#####################
+# Iterator protocol #
+#####################
+
+
+@ovld
+def myia_iter(obj: range):
+    """Create an iterator."""
+    return obj
+
+
+@ovld
+def myia_iter(obj: tuple):  # noqa: F811
+    return obj
+
+
+@ovld
+def myia_hasnext(obj: range):
+    """Return whether the iterator has a next production."""
+    return obj.start < obj.stop
+
+
+@ovld
+def myia_hasnext(obj: tuple):  # noqa: F811
+    return bool(obj)
+
+
+@ovld
+def myia_next(obj: range):
+    """Return the iterator's next production and the next iterator."""
+    return obj.start, range(obj.start + obj.step, obj.stop, obj.step)
+
+
+@ovld
+def myia_next(obj: tuple):  # noqa: F811
+    return obj[0], obj[1:]
+
+
+##############################
+# Handle and global universe #
+##############################
+
+
+class Handle:
+    """Handle class for `make_handle`."""
+
+    def __init__(self, object_type, value=None):
+        self.type = object_type
+        self.value = value
+
+
+def make_handle(object_type):
+    """Create a handle of the given type."""
+    return Handle(object_type)
+
+
+def global_universe_getitem(h):
+    """Set the value of a handle."""
```

Comment:
I think it should be "Get the value of a handle" here.

Reply:
Oh, right, haha.